cabal-install/src/Distribution/Client/ProjectPlanning.hs

```diff
@@ -397,13 +398,13 @@ rebuildProjectConfig
           localPackages <- phaseReadLocalPackages (projectConfig <> cliConfig)
           return (projectConfig, localPackages)
 
-    sequence_
-      [ do
-        notice verbosity . render . vcat $
-          text "Configuration is affected by the following files:"
-            : [text "-" <+> docProjectConfigPath path]
-      | Explicit path <- Set.toList $ projectConfigProvenance projectConfig
-      ]
+    let configfiles =
+          [ text "-" <+> docProjectConfigPath path
```

Comment:
Not a big fan of list comprehensions, but I understand this was not originally written by you and this is a time sensitive PR.

Reply:
I'm happy to improve it. Do you have suggestions?

Reply:
Like, it'd be easy to rewrite it with `do` but I'm not sure that's better. And if we wanna stay away from syntactic sugar altogether, this looks like a rather ugly fold to me.

Reply:
I see now we are filtering out `Implicit`. It is OK then!

Reply:
Yes!